yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -395,6 +395,10 @@ private[yarn] class YarnAllocator(
       val executorId = executorIdCounter.toString
 
       assert(container.getResource.getMemory >= resource.getMemory)
```

Comment:
Not sure from Yarn side the granted memory will possibly be less than the requested, I haven't met such problem yet.